cmd/fs/helpers.go

```diff
@@ -46,6 +48,116 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+const volumesPefix string = "dbfs:/Volumes"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(
+	pathArgCount int,
+	onlyDirs bool,
+	filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error),
+	mustWorkspaceClientFunc func(cmd *cobra.Command, args []string) error,
+) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClientFunc(cmd, args)
+
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isDbfsPath := isDbfsPath(toComplete)
+
+		wsc := root.WorkspaceClient(cmd.Context())
+		_, err = wsc.CurrentUser.Me(cmd.Context())
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		completer := completer.NewCompleter(cmd.Context(), filer, onlyDirs)
+
+		if len(args) >= pathArgCount {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		completions, dirPath, directive := completer.CompletePath(path)
+
+		for i := range completions {
+			completions[i] = prependDirPath(dirPath, completions[i], !isDbfsPath)
+
+			if isDbfsPath {
+				// The dirPath will start with a "/", so we'll prefix the completions with
+				// the selectedPrefix without a trailing "/"
+				prefix := dbfsPrefix[:len(dbfsPrefix)-1]
+				completions[i] = fmt.Sprintf("%s%s", prefix, completions[i])
```

Comment:
In this case we don't want to do any cleanup of the paths as the user might be enter a strange paths //a/b///c that needs to be completed. There's a comment about it on line 125.

Reply:
That's fine! We end up cleaning the path in a bunch of place anyways, so cleaning them up before autocomplete makes sense. 

For example, you can try running:
```
databricks fs ls ~/////////
```

Reply:
The problem is it breaks the autocomplete because our completion script will match the suggestions against what the user has typed on the command line. If the user types dbfs://foo and autocompletion returns dbfs:/foo, there won't be any suggestions because they don't match.

Reply:
Will use `path.Join`! (see https://github.com/databricks/cli/pull/1622#discussion_r1704214774)

Reply:
Really? That sucks. Is there a way around it? With the current approach, it's on the API to tolerate non-clean paths, and then return a valid response to them upon listing. 

Does dbfs://foo work right now and give valid autocomplete suggestions?

Reply:
It worked until I migrated to using `path` and `filepath` for joining paths :D

The workaround would be manually handle it